### PR TITLE
chore: SonarQube added in circleci pipeline

### PR DESCRIPTION
**Pull request by aryamohanan**

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# Code coverage configuration
+sonar.javascript.lcov.reportPaths=coverage/lcov.info
+
+# ESLint configuration
+sonar.eslint.reportPaths=eslint-report.json
+
+# Code analysis parameters  
+sonar.inclusions=**/src/**,**/lib/**
+sonar.exclusions=**/test/**,**/dummy-app/**,**/precompile/**.,**/uncaught/**
+sonar.coverage.exclusions=**/test/**,**/dummy-app/**,**/precompile/**,**/uncaught/**
```
Comment (basti1302) [inline sonar-project.properties:30]: I understand all exclusions except for the `**/uncaught/**` one. That would exclude `packages/collector/src/uncaught` from coverage analysis as well as from normal static code analysis, right? Why would we want to exclude that?

**Review by basti1302 — APPROVED** [review]
Awesome!

File: .circleci/config.yml [hunk 2]
```diff
@@ -279,6 +279,20 @@ shared: &shared
     - run:
         name: Analyze Test Durations
         command: node misc/test-durations/ci
+    # Generate ESLint report
+    - run:
+        name: Run ESLint and generate ESLint report
+        command: |
+          if [ -n "$NODE_COVERAGE" ]; then
+            npx eslint packages/ -f json > eslint-report.json || exit 0
```
[the remark this answers is not in the document]
Comment (aryamohanan) [reply .circleci/config.yml:287]: Thank you for bringing this up. When it comes to SonarQube, it requires the ESLint report to be in JSON format, which is why I included the -f json flag. The existing step, on the other hand, performs additional tasks such as running TypeScript (tsc) and using ESLint via lerna. These additional tasks aren't necessary for SonarQube, and the existing step is common to other jobs as well. Therefore, I thought it might be a good idea to isolate this command exclusively for SonarQube. I'm open to alternative suggestions, so please feel free to share your thoughts as well.

https://docs.sonarsource.com/sonarqube/latest/analyzing-source-code/importing-external-issues/importing-third-party-issues/#:~:text=sonar.eslint.reportPaths

File: sonar-project.properties (hunk 1 above)
Comment (aryamohanan) [reply sonar-project.properties:29]: The code coverage for the file `syncHttp.js` in this particular folder was 0, that's why I excluded this from the analysis, Removing it from here will not have a significant impact on the overall coverage.

File: sonar-project.properties [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Project identification
+sonar.projectKey=nodejs-tracer
+sonar.projectName=NodeJS Tracer
+sonar.projectVersion=1.0
+sonar.host.url=https://sonarqube.instana.io
+sonar.token=${{ secrets.SONAR_TOKEN }}
+
+# Source code locations
+sonar.sources=packages
+sonar.scm.disabled=true
```
[the remark this answers is not in the document]
Comment (aryamohanan) [reply sonar-project.properties]: The sonar.scm.disabled=true setting disables SonarQube's integration with source control management (SCM) systems. When this setting is enabled, SonarQube will not attempt to retrieve any information from the SCM system, such as the authors of changes, the revision history of files, or the list of ignored files. I will remove this from the settings and will check
https://docs.sonarsource.com/sonarqube/latest/analyzing-source-code/analysis-parameters/#:~:text=always%20disabled%20if-,sonar.scm.disabled,-is%20set%20to

File: sonar-project.properties (hunk 1 above)
Comment (basti1302) [reply sonar-project.properties]: I would suggest not excluding it, since it is part of our production code. Excluding it feels a bit like cheating. If it is uncovered, it is uncovered, so be it.

Side note: For some reason `syncHttp.js` does not show up at all in our HTML coverage report https://output.circle-artifacts.com/output/job/dd17c749-4aef-4335-a1c3-497ed8f98fc3/artifacts/0/coverage/lcov-report/collector/src/uncaught/index.html, I wonder why... 

Oh, wait! I think `syncHttp.js` file is completely unused and we can actually delete it! That's why it also has 0% test coverage :D 

Looks like we removed its usage here: https://github.com/instana/nodejs/commit/fb6570a862dbdec776eb78b840dcdc4184cd5f00#diff-8ff40b8dde5d139718e5cd429da1d02c4aac512218fa125e3947c0b0e049df8cL481 but forgot to delete the file itself. We can delete that file in a different PR or in this PR, whatever you prefer.

Cool, so Sonar already uncovered an actual code quality issue. Yay, Sonar!

Okay, long story short, let's remove `**/uncaught/**` from both `sonar.exclusions` and `sonar.coverage.exclusions`.

File: sonar-project.properties (hunk 3 above)
[the remark this answers is not in the document]
Comment (aryamohanan) [reply sonar-project.properties]: The property updated to sonar.scm.exclusions.disabled

[the remark this answers is not in the document]
Comment (basti1302) [reply sonar-project.properties]: Hm, not sure about that. I also didn't mean to suggest that change :-)  I just wondered why we need to disable SCM integration in the first place.

Now for `sonar.scm.exclusions.disabled`, I don't think we need this, do we? By default, files in `.gitingore` are ignored by Sonar, which seems to be the correct behaviour, no? And when we set this setting, files in `.gitignore` are no longer ignored by Sonar.
